### DISCUSSION TITLE
onFirstEnter and onFirstLeave

### DISCUSSION THREAD

**Discussion by shachar731945** [petyosi/react-virtuoso#355] [Ideas]
I would like to add a feature regarding the first time an item enters or leaves the view.

An example of why it is helpful is a messaging an app that when an app first enters into view, it will send a notification to the server that the message has been read.

Is it something that would be viable in the package's view?

I would like to add the 2 function as props for the Virtuoso component and it will pass the index and the item (if the data prop exists) to the function

**Comment by petyosi** [comment]
The use case you describe makes sense to me. I don't think that you need to make this a built-in behavior, as there are several tweaks which better be left to the developer to decide (for example - use timeout to decide if the message has been "properly" seen). 

As a starting point, you have everything you need available in the `rangeChanged` callback - you can start from there.
